pkg/catalog/manager/manager.go

```diff
@@ -243,24 +241,38 @@ func (m *Manager) ValidateKubeVersion(template *v3.CatalogTemplateVersion, clust
 }
 
 func (m *Manager) ValidateRancherVersion(template *v3.CatalogTemplateVersion) error {
-	rancherMin := template.Spec.RancherMinVersion
-	rancherMax := template.Spec.RancherMaxVersion
-
 	serverVersion := settings.ServerVersion.Get()
 
 	// don't compare if we are running as dev or in the build env
 	if !utils.ReleaseServerVersion(serverVersion) {
 		return nil
 	}
 
-	if rancherMin != "" && !mVersion.Compare(serverVersion, rancherMin, ">=") {
-		return fmt.Errorf("rancher min version not met")
-	}
+	serverVersion = strings.TrimPrefix(serverVersion, "v")
 
-	if rancherMax != "" && !mVersion.Compare(serverVersion, rancherMax, "<=") {
-		return fmt.Errorf("rancher max version exceeded")
+	versionRange := ""
+	if template.Spec.RancherMinVersion != "" {
+		versionRange += " >=" + strings.TrimPrefix(template.Spec.RancherMinVersion, "v")
+	}
+	if template.Spec.RancherMaxVersion != "" {
+		versionRange += " <=" + strings.TrimPrefix(template.Spec.RancherMaxVersion, "v")
+	}
+	if versionRange == "" {
+		return nil
+	}
+	constraint, err := semver.ParseRange(versionRange)
+	if err != nil {
+		logrus.Errorf("failed to parse constraint for rancher version %s: %v", versionRange, err)
+		return nil
 	}
 
+	rancherVersion, err := semver.Parse(serverVersion)
+	if err != nil {
+		return err
```

Comment:
Seems odd to treat these differently, I think it may make sense to log and swallow both since it is probably more annoying to not be able to install a catalog you should be able to install then it is bad that you can install a catalog that you shouldn't be able to install. I do suppose an invalid upgrade could be kind of bad though. @snasovich thoughts?

Reply:
This was how it is being done for the kubernetes version, but tI think that makes sense since the only way it will fail to parse is if the server version has been manually set.

Reply:
Yes, it matches the following checks so I thought this inconsistency is OK when I reviewed: https://github.com/rancher/rancher/blob/e013338b5eb1271ced81b7ff7ebae093f664d7dc/pkg/catalog/manager/manager.go#L218-L236

I could see some logic in it - if there is the first error, it's an issue with constraints on `CatalogTemplateVersion` so we err on the side of "let's allow to install it" while the second error is when Rancher version is messed up and we block installs against it.

Since I don't know the history of why error handling is inconsistent and there is not a comment that explains it I find it safer to follow the same logic here too.

Reply:
@snasovich yeah that makes sense.